### ISSUE TITLE
Define schemas in a separate package

### ISSUE THREAD

**Issue by rrousselGit** [gql-dart/ferry#604]
Hello!

For the sake of sharing code between backend and frontend, I'd like to define my schema files outside of the package that uses ferry_generator.

Would it be possible to support in the `build.yaml` a `schema` that comes from a different package?

**Comment by knaeckeKami** [comment]
Hi!

Just for clarification:

You want to be able to do something like

- have a separate package that contains the graphql schema
- a separate fronted and backend package, which both depend on the `schema` package, but their own graphql operations

Is this correct? 

I'll look into supporting this, should be quite easy, unless there is an unexpected issue with built_value serialization

**Comment by rrousselGit** [comment]
That is correct, yes.

**Comment by knaeckeKami** [comment]
a draft PR is available at #606.

still needs a cleanup and tests, but you can already try it out by using 

```yaml

ferry_generator:
   git:
     url: https://github.com/gql-dart/ferry.git
     path: packages/ferry_generator
     ref: external_schema_import 
```

**Comment by knaeckeKami** [comment]
this was released in ferry_generator 0.12.0
